js/user-form.js

```diff
@@ -0,0 +1,155 @@
+
+const form = document.getElementById('form');
+const btn = document.getElementById('submission');
+
+// Error Handler
+function showError(input, message) {
+  const formGroup = input.parentElement;
+  formGroup.className = 'form-group error';
+  const small = formGroup.querySelector('small');
+  small.innerText = message;
+}
+
+function showSuccess(input) {
+  const formGroup = input.parentElement;
+  formGroup.className = 'form-group success';
+}
+
+function checkUsername(input) {
+  //Check presence of first_name
+  if (!input.value.trim().startsWith(first_name.value.toLowerCase())) {
+    showError(
+      username,
+      'Username must start with first name. Usage of funky words should be avoided.'
+    );
+  } else {
+    showSuccess(username);
+  }
+
+  // Check if username pertains to the appropriate format
+  let pattern = new RegExp(/^[a-zA-Z-]+$/g);
+  if (pattern.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(
+      username,
+      'Only alphabets and "-" are allowed. Numbers are prohibited.'
+    );
+  }
+}
+
+function checkEmail(input) {
+  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (re.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(input, 'Email is not valid');
+  }
+}
+
+function checkPhone(input) {
+  const re = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
+  if (re.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(input, 'Phone number is not valid');
+  }
+}
+
+//Input fields validator
+function checkRequired(inputKeys) {
+  const arrEls = getElementsForKeys(inputKeys);
+  arrEls.forEach((input) => {
+    if (input.value.trim() === '') {
+      showError(input, 'Field cannot be empty.');
+      btn.disabled = true;
+    } else {
+      showSuccess(input);
+      btn.disabled = false;
+      btn.style.opacity = 1;
+    }
+  });
+}
+
+//Form submission handler
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  checkUsername(username);
+  checkEmail(email);
+  checkPhone(phone);
+  checkRequired([
+    USERNAME,
+    FIRST_NAME,
+    LAST_NAME,
+    YOE,
+    COMPANY,
+    DESIGNATION,
+    GITHUB_ID,
+    LINKEDIN_ID,
+    TWITTER_ID
+  ]);
+
+  updateUserData();
+});
+
+function getDataToSend (arrKeys) {
+  return arrKeys.reduce(function(acc, k) {
+    const el = document.getElementById(k);
+
+    if(el && el.tagName === 'INPUT') {
+      acc[k] = el.value;
+    }
+    return acc;
+  }, {})
+}
+
+const updateUserData = async () => {
+  
+  const dataToSend = getDataToSend([
+    USERNAME,
+    FIRST_NAME,
+    LAST_NAME,
+    EMAIL,
+    PHONE, 
+    YOE,
+    COMPANY,
+    DESIGNATION,
+    GITHUB_ID,
+    LINKEDIN_ID,
+    TWITTER_ID
+  ]);
+
+  console.log(dataToSend);
+
+  try {
+    const res = await fetch(`AWAITING_API__https://staging-api.realdevsquad.com/users/self`, {
```

Comment:
@Rucha1499 I am ready to start using the API. Let me know when it's ready please